[gtk3] Unable to build

```
The Meson build system
Version: 0.55.3
Source dir: /usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23
Build dir: /usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/build
Build type: native build
WARNING: Unknown options: "LIB_INSTALL_DIR, SYSCONFDIR"
The value of new options can be set with:
meson setup <builddir> --reconfigure -Dnew_option=new_value ...
Project name: gtk+-3.0
Project version: 3.24.23
Using 'CFLAGS' from environment with value: '-fuse-ld=lld'

meson.build:1:0: ERROR: Unable to determine dynamic linker

A full log can be found at /usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/build/meson-logs/meson-log.txt
gtk3 failed to build: `meson -Dbuildtype=release -Dbroadway_backend=true -Dgtk_doc=false -Ddemos=false -Dexamples=false -Dprefix=/usr/local -Dlibdir=/usr/local/lib -DLIB_INSTALL_DIR=/usr/local/lib -Dmandir=/usr/local/share/man -DSYSCONFDIR=/usr/local/etc -Ddatadir=/usr/local/lib -Dbuildtype=release build` exited with 1
```

Also on x86_64:
```
FAILED: gdk/Gdk-3.0.gir 
/usr/local/bin/g-ir-scanner --no-libtool --namespace=Gdk --nsversion=3.0 --warn-all --output gdk/Gdk-3.0.gir --c-include=gdk/gdk.h --quiet -DGDK_COMPILATION -I/usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/gdk -I/usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/build/gdk -I./. -I../. -I./gdk/x11 -I../gdk/x11 -I./gdk/. -I../gdk/. --filelist=/usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/build/gdk/libgdk-3.so.0.2404.19.p/Gdk_3.0_gir_filelist --include=cairo-1.0 --include=Gio-2.0 --include=GdkPixbuf-2.0 --include=Pango-1.0 --symbol-prefix=gdk --identifier-prefix=Gdk --pkg-export=gdk-3.0 --cflags-begin -DG_LOG_USE_STRUCTURED=1 -DGLIB_MIN_REQUIRED_VERSION=GLIB_VERSION_2_56 -DGLIB_MAX_ALLOWED_VERSION=GLIB_VERSION_2_58 '-DGTK_VERSION="3.24.23"' -D_GNU_SOURCE -DG_DISABLE_CAST_CHECKS -I./. -I../. -I./gdk/x11 -I../gdk/x11 -I./gdk/. -I../gdk/. -I/usr/local/include/gdk-pixbuf-2.0 -I/usr/local/include/libmount -I/usr/local/include/blkid -I/usr/local/lib64/libffi-3.2.1/include -I/usr/local/include/glib-2.0 -I/usr/local/lib64/glib-2.0/include -I/usr/local/include/cairo -I/usr/local/include/pixman-1 -I/usr/local/include/uuid -I/usr/local/include/freetype2 -I/usr/local/include/libdrm -I/usr/local/include/libpng16 -I/usr/local/include/pango-1.0 -I/usr/local/include/fribidi -I/usr/local/include/harfbuzz -I/usr/local/include/gio-unix-2.0 -I/usr/local/include/gobject-introspection-1.0 --cflags-end --add-include-path=/usr/local/share/gir-1.0 --library gdk-3 -L/usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/build/gdk --extra-library=m --extra-library=gdk_pixbuf-2.0 --extra-library=gobject-2.0 --extra-library=glib-2.0 --extra-library=cairo --extra-library=pango-1.0 --extra-library=harfbuzz --extra-library=fribidi --extra-library=cairo-gobject --extra-library=epoxy --extra-library=fontconfig --extra-library=freetype --extra-library=gio-2.0 --extra-library=pangocairo-1.0 --extra-library=rt --extra-library=xkbcommon --extra-library=wayland-client --extra-library=wayland-cursor --extra-library=wayland-egl --extra-library=Xrender --extra-library=X11 --extra-library=Xi --extra-library=Xext --extra-library=Xcursor --extra-library=Xdamage --extra-library=Xfixes --extra-library=Xcomposite --extra-library=Xrandr --extra-library=Xinerama --extra-library=girepository-1.0 --sources-top-dirs /usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/subprojects/ --sources-top-dirs /usr/local/tmp/crew/gtk+-3.24.23.tar.xz.dir/gtk+-3.24.23/build/subprojects/
Couldn't find include 'Pango-1.0.gir' (search path: '['/usr/local/share/gir-1.0', '/usr/local/share', '/usr/local/share/gir-1.0', '/usr/local/share/gir-1.0', '/usr/share/gir-1.0']')
[160/1198] Generating gtkprivatetypebuiltins.c with a meson_exe.py custom command
[161/1198] Generating gtktypebuiltins.h with a meson_exe.py custom command
ninja: build stopped: subcommand failed.
gtk3 failed to build: `ninja -C build` exited with 1
```

```
Couldn't find include 'Pango-1.0.gir' (search path: '['/usr/local/share/gir-1.0', '/usr/local/share', 'gir-1.0', '/usr/local/share/gir-1.0', '/usr/share/gir-1.0', '/usr/local/share/gir-1.0', '/usr/share/gir-1.0']')
[183/1198] Compiling C object gtk/libgtk-3.so.0.2404.19.p/meson-generated_.._gtkmarshalers.c.o
ninja: build stopped: subcommand failed.
```

From crew files pango:
```
/usr/local/lib64/gir-1.0/Pango-1.0.gir
/usr/local/lib64/gir-1.0/PangoCairo-1.0.gir
/usr/local/lib64/gir-1.0/PangoFc-1.0.gir
/usr/local/lib64/gir-1.0/PangoFT2-1.0.gir
/usr/local/lib64/gir-1.0/PangoOT-1.0.gir
/usr/local/lib64/gir-1.0/PangoXft-1.0.gir
```
I think the problem here is at https://github.com/skycocker/chromebrew/blob/3f03c8eb6f4053b79d0638e97bd7e59620665826/lib/const.rb#L64
Note how datadir is set to ```-Ddatadir=#{CREW_LIB_PREFIX}``` - it needs to be set to ```-Ddatadir=#{CREW_PREFIX}/share```

This affects harfbuzz too for the same reason...

--- a/packages/gtk3.rb
+++ b/packages/gtk3.rb
@@ -30,20 +30,7 @@ class Gtk3 < Package
     # The lld linker allows linking against system ChromeOS libs.
     ENV['CFLAGS'] = "-fuse-ld=lld"
     ENV['CXXFLAGS'] = "-fuse-ld=lld"
-    system "meson",
-         '-Dbuildtype=release',
-         '-Dbroadway_backend=true',
-         '-Dgtk_doc=false',
-         '-Ddemos=false',
-         '-Dexamples=false',
-         "-Dprefix=#{CREW_PREFIX}",
-         "-Dlibdir=#{CREW_LIB_PREFIX}",
-         "-DLIB_INSTALL_DIR=#{CREW_LIB_PREFIX}",
-         "-Dmandir=#{CREW_MAN_PREFIX}",
-         "-DSYSCONFDIR=#{CREW_PREFIX}/etc",
-         "-Ddatadir=#{CREW_LIB_PREFIX}",
-         '-Dbuildtype=release',
-         'build'
+    system "meson #{CREW_MESON_OPTIONS} -Dbroadway_backend=true -Dgtk_doc=false -Ddemos=false -Dexamples=false build"
     system "meson configure build"
     system 'ninja -C build'
   end

